enderio-machines/src/main/java/com/enderio/machines/common/blockentity/XPVacuumBlockEntity.java

```diff
@@ -0,0 +1,130 @@
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.world.entity.ExperienceOrb;
+import net.minecraft.world.entity.player.Inventory;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.inventory.AbstractContainerMenu;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.entity.BlockEntityType;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.material.Fluids;
+import net.minecraft.world.phys.AABB;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.capability.IFluidHandler.FluidAction;
+
+public class XPVacuumBlockEntity extends MachineBlockEntity {
+    private static final double COLLISION_DISTANCE_SQ = 1 * 1;
+    private static final double SPEED = 0.025;
+    private static final double SPEED_4 = SPEED*4 ;
+    private static final int MAX_RANGE = 6;
+    private int range = 6;
+    private FluidTankMaster fluidTank;
+    private List<WeakReference<ExperienceOrb>> xpEntities = new ArrayList<>();
+    
+    public XPVacuumBlockEntity(BlockEntityType<?> pType, BlockPos pWorldPosition, BlockState pBlockState) {
+        super(MachineTier.STANDARD, pType, pWorldPosition, pBlockState);
+        this.fluidTank =  new FluidTankMaster(Integer.MAX_VALUE, getIoConfig()); //that seems quite large?
+        addDataSlot(new IntegerDataSlot(() -> fluidTank.getFluidInTank(0).getAmount(), (i) -> fluidTank.setFluid(new FluidStack(Fluids.WATER, i)), SyncMode.WORLD));
+        add2WayDataSlot(new IntegerDataSlot(() -> this.range, this::setRange, SyncMode.GUI));
+    }
+    
+    public FluidTankMaster getFluidTank() {
+        return fluidTank;
+    }
+    
+    @Override
+    public void saveAdditional(CompoundTag pTag) {
+        super.saveAdditional(pTag);
+        // TODO: Rename to fluid or tank.
+        // TODO: Common place for all NBT names.
+        pTag.put("Fluids", fluidTank.writeToNBT(new CompoundTag()));
+    }
+    
+    @Override
+    public void load(CompoundTag pTag) {
+        super.load(pTag);
+        fluidTank.readFromNBT(pTag.getCompound("Fluids"));
+    }
+    
+    @Override
+    public AbstractContainerMenu createMenu(int containerId, Inventory inventory, Player player) {
+        return new XPVacuumMenu(this, inventory, containerId);
+    }
+    
+    @Override
+    public void tick() {
+        if (this.getRedstoneControl().isActive(level.hasNeighborSignal(worldPosition))) {
+            this.collectXP(this.getLevel(), this.getBlockPos(), this.range);
+        }
+        super.tick();
+    }
+    
+    private void collectXP(Level level, BlockPos pos, int range) {
+        if ((level.getGameTime() + pos.asLong()) % 5 == 0) {
+            AABB area = new AABB(pos).inflate(range);
+            for (ExperienceOrb ex:level.getEntitiesOfClass(ExperienceOrb.class, area)) {
+                this.xpEntities.add(new WeakReference<ExperienceOrb>(ex));
+            };
+        }
```

Comment:
everything from VacuumChestBlockEntity
create common and generic helper for this and the itemEntity finding
parameters you probably will use: list, class, level, pos, range, filter (Predicate)